diesel/src/pg/types/mod.rs

```diff
@@ -88,4 +90,55 @@ pub mod sql_types {
 
     #[doc(hidden)]
     pub type Bpchar = ::types::VarChar;
+
+    #[cfg(feature = "serde_json")]
+    /// The JSON SQL type.  This type can only be used with `feature =
+    /// "serde_json"`
+    ///
+    /// Normally you should prefer `Jsonb` instead, for the reasons
+    /// discussed there.
+    ///
+    /// ### [`ToSql`](/diesel/types/trait.ToSql.html) impls
+    ///
+    /// - [`serde_json::Value`][Value]
+    ///
+    /// ### [`FromSql`](/diesel/types/trait.FromSql.html) impls
+    ///
+    /// - [`serde_json`][Value]
+    ///
+    /// [Value]: https://docs.serde.rs/serde_json/value/enum.Value.html
+    #[derive(Debug, Clone, Copy, Default)] pub struct Json;
+
+    #[cfg(feature = "serde_json")]
+    /// The `jsonb` SQL type.  This type can only be used with `feature =
+    /// "serde_json"`
+    ///
+    /// `jsonb` offers [several advantages][adv] over regular JSON:
+    ///
+    /// > There are two JSON data types: `json` and `jsonb`. They accept almost
+    /// > identical sets of values as input. The major practical difference
+    /// > is one of efficiency. The `json` data type stores an exact copy of
+    /// > the input text, which processing functions must reparse on each
+    /// > execution; while `jsonb` data is stored in a decomposed binary format
+    /// > that makes it slightly slower to input due to added conversion
+    /// > overhead, but significantly faster to process, since no reparsing
+    /// > is needed. `jsonb` also supports indexing, which can be a significant
+    /// > advantage.
+    /// >
+    /// > ...In general, most applications should prefer to store JSON data as
+    /// > `jsonb`, unless there are quite specialized needs, such as legacy
+    /// > assumptions about ordering of object keys.
+    ///
+    /// [adv]: https://www.postgresql.org/docs/9.6/static/datatype-json.html
+    ///
+    /// ### [`ToSql`](/diesel/types/trait.ToSql.html) impls
+    ///
+    /// - [`serde_json::Value`][Value]
+    ///
+    /// ### [`FromSql`](/diesel/types/trait.FromSql.html) impls
+    ///
+    /// - [`serde_json`][Value]
+    ///
+    /// [Value]: https://docs.serde.rs/serde_json/value/enum.Value.html
+    #[derive(Debug, Clone, Copy, Default)] pub struct Jsonb;
```

Comment:
Do you think we should call this type `JsonB` and have `#[doc(hidden)] pub type Jsonb = JsonB` to make schema inference happy [like we do for VarChar](https://github.com/diesel-rs/diesel/blob/7998aa351f3c591ccb2d9a1a3a9cfd2737f91727/diesel/src/types/mod.rs#L153)?